src/AuthProvider.tsx

```diff
@@ -74,32 +74,42 @@ export interface AuthProviderPropsBase extends UserManagerSettings {
      * @deprecated On sign out popup hook. Can be a async function.
      */
     onSignoutPopup?: () => Promise<void> | void;
+}
 
+/**
+ * This interface (default) is used to pass `UserManagerSettings` together with `AuthProvider` properties to the provider.
+ *
+ * @public
+ */
+export interface AuthProviderNoUserManagerProps extends AuthProviderBaseProps, UserManagerSettings
+{
     /**
-     * Allow passing a custom UserManager.
+     * @deprecated Allow passing a custom UserManager implementation
      */
-    userManager?: UserManager;
+    implementation?: typeof UserManager | null;
 
     /**
-     * @deprecated Allow passing a custom UserManager implementation
+     * Prevent this property.
      */
-    implementation?: typeof UserManager | null;
+    userManager?: never;
 }
 
 /**
+ * This interface is used to pass directly a `UserManager` instance together with `AuthProvider` properties to the provider.
+ *
  * @public
  */
-export interface AuthProviderUserManagerProps extends Omit<AuthProviderPropsBase, "redirect_uri" | "client_id" | "authority"> {
-    redirect_uri?: never;
```

Comment:
shouldn't we also ensure that these are `never` so that you can't pass in a UserManager and a redirect uri?

Reply:
The main change i made was that new: `AuthProviderUserManagerProps` no longer extends `UserManagerSettings`, thus you will get a type error when tying to combine the userManager and something from `UserManagerSettings` including e.g. `redirect_uri`.